src/js/services/link_track.js

```diff
@@ -0,0 +1,186 @@
+    eventAction: 'click',
+    eventLabel: link.href,
+  };
+}
+
+/**
+ * Computes the data for a given anchor element
+ *
+ * @param {HTMLAnchorElement} link
+ * @return {Object} object with the values to call ga send event command.
+ */
+function getData(link) {
+  const data = Object.assign(
+    defaultsForLink(link),
+    link.dataset,
+    compulsoryFields(link),
+  );
+
+  return EVENT_FIELDS.reduce((memo, key) => {
+    /* eslint no-param-reassign: ["error", { "props": false }] */
+    if (key in data) {
+      memo[key] = data[key];
+    }
+    return memo;
+  }, {});
+}
+
+/**
+ * Returns the arguments to call ga command when the link has been clicked.
+ *
+ * @param {HTMLAnchorElement} link element to get the data from.
+ * @return {Array<String, String, Object>} arguments for ga method.
+ */
+function commandArguments(link) {
+  return [SEND_COMMAND, EVENT_HIT_TYPE, getData(link)];
+}
+
+const gaInternal = [];
+const ga = window.ga || function (...args) {
+  gaInternal.push(args);
+  console.log(...args);
+
+  const last = args.pop();
+
+  if (typeof last.hitCallback === 'function') {
+    setTimeout(last.hitCallback.bind(last), 0);
+  }
+};
+
+/**
+ * Promisify ga
+ *
+ * @param {...any} args
+ * @return {Promise} a promise that resolves whenever the hitCallback is called.
+ */
+function gaPromise(...args) {
+  let obj = {};
+  let oldCallback = () => { };
+
+  if (typeof args[args.length - 1] === 'object') {
+    obj = args.pop();
+  }
+
+  if (typeof obj.hitCallback === 'function') {
+    oldCallback = obj.hitCallback;
+  }
+
+  return new Promise((resolve) => {
+    obj.hitCallback = (...cbArgs) => {
+      oldCallback(...cbArgs);
+      resolve(...cbArgs);
+    };
+    ga(...args, obj);
+  });
+}
+
+function isScrollableLink(link) {
+  const [url, id] = link.href.split('#');
+
+  return window.location.href.indexOf(url) > 0 && id;
+}
+
+/**
+ * Scrolls to element anchored by the given link.
+ *
+ * @param {HTMLAnchorElement} link
+ */
+function scrollTo(link) {
+  const [, id] = link.href.split('#');
+  $('body, html').animate(
+    {
+      scrollTop: $(`#${id}`).offset().top,
+    },
+    1000,
+  );
+}
+
+/**
+ * Sends to Google Analytics the details from the link
+ *
+ * @param {HTMLAnchorElement} link anchor clicked
+ * @return {Promise} a promise resolved whenever hitCallback is called.
+ */
+export function sendLinkDetails(link) {
+  return gaPromise(...commandArguments(link));
+}
+
+/**
+ * Emulates the click on the link.
+ *
+ * @param {HTMLAnchorElement} link anchor to emulate
+ */
+export function goToLink(link) {
+  if (link.target === '_blank') {
+    window.open(link.href);
+  } else {
+    window.location.href = link.href;
+  }
+}
+
+/**
+ * Sets up the tracking of links, inbound and outbound.
+ *
+ * For a link to be tracked it needs to:
+ *
+ * 1. Have a `data-tracked` attribute.
+ * 2. Have a non-empty `href` attribute.
+ */
+export default function setupLinkTracking() {
```

Comment:
I'll check our configuration, but usually linters complain about things not being defined, and that includes called functions.